temporalcli/commands.workflow_test.go

```diff
@@ -114,3 +114,100 @@ func (s *SharedServerSuite) testSignalBatchWorkflow(json bool) *CommandResult {
 	}
 	return res
 }
+func (s *SharedServerSuite) TestWorkflow_Terminate_SingleWorkflowSuccess() {
+	// Make workflow wait for termination and then return the context's error
+	s.Worker.OnDevWorkflow(func(ctx workflow.Context, a any) (any, error) {
+		ctx.Done().Receive(ctx, nil)
+		return nil, ctx.Err()
+	})
+
+	// Start the workflow
+	run, err := s.Client.ExecuteWorkflow(
+		s.Context,
+		client.StartWorkflowOptions{TaskQueue: s.Worker.Options.TaskQueue},
+		DevWorkflow,
+		"ignored",
+	)
+	s.NoError(err)
+
+	// Send terminate
+	res := s.Execute(
+		"workflow", "terminate",
+		"--address", s.Address(),
+		"-w", run.GetID(),
+		// Ensure that we can provide a reason
+		"--reason", "terminate-test",
+	)
+	s.NoError(res.Err)
+
+	// Confirm workflow was terminated
+	s.Contains(run.Get(s.Context, nil).Error(), "terminated")
+}
+
+func (s *SharedServerSuite) TestWorkflow_Terminate_BatchWorkflowSuccess() {
+	res := s.testTerminateBatchWorkflow(false)
+	s.Contains(res.Stdout.String(), "approximately 5 workflow(s)")
+	s.Contains(res.Stdout.String(), "Started batch")
+}
+
+func (s *SharedServerSuite) TestWorkflow_Terminate_BatchWorkflowSuccessJSON() {
+	res := s.testTerminateBatchWorkflow(true)
+	var jsonRes map[string]any
+	s.NoError(json.Unmarshal(res.Stdout.Bytes(), &jsonRes))
+	s.NotEmpty(jsonRes["batchJobId"])
+}
+
+func (s *SharedServerSuite) testTerminateBatchWorkflow(json bool) *CommandResult {
+	// Make workflow wait for terminate and then return the context's error
+	s.Worker.OnDevWorkflow(func(ctx workflow.Context, a any) (any, error) {
+		ctx.Done().Receive(ctx, nil)
+		return nil, ctx.Err()
+	})
+
+	// Start 5 workflows
+	runs := make([]client.WorkflowRun, 5)
+	searchAttr := "keyword-" + uuid.NewString()
+	for i := range runs {
+		run, err := s.Client.ExecuteWorkflow(
+			s.Context,
+			client.StartWorkflowOptions{
+				TaskQueue:        s.Worker.Options.TaskQueue,
+				SearchAttributes: map[string]any{"CustomKeywordField": searchAttr},
+			},
+			DevWorkflow,
+			"ignored",
+		)
+		s.NoError(err)
+		runs[i] = run
+	}
+
+	// Wait for all to appear in list
+	s.Eventually(func() bool {
+		resp, err := s.Client.ListWorkflow(s.Context, &workflowservice.ListWorkflowExecutionsRequest{
+			Query: "CustomKeywordField = '" + searchAttr + "'",
+		})
+		s.NoError(err)
+		return len(resp.Executions) == len(runs)
+	}, 3*time.Second, 100*time.Millisecond)
+
+	// Send batch terminate with a "y" for non-json or "--yes" for json
+	args := []string{
+		"workflow", "terminate",
+		"--address", s.Address(),
+		"--query", "CustomKeywordField = '" + searchAttr + "'",
+		"--reason", "terminate-test",
+	}
+	if json {
+		args = append(args, "--yes", "-o", "json")
+	} else {
+		s.CommandHarness.Stdin.WriteString("y\n")
+	}
+	res := s.Execute(args...)
+	s.NoError(res.Err)
+
+	// Confirm that all workflows are terminated
+	for _, run := range runs {
+		s.Contains(run.Get(s.Context, nil).Error(), "terminated")
```

Comment:
Same here, would be nice to verify the reason was propagated.

Reply:
Done! Thanks for reminding me to do so